### ISSUE TITLE
If I back out of a proof request, that proof erroneously shows up on the NEXT QR code I try.

### ISSUE THREAD

**Issue by loneil** [bcgov/bc-wallet-mobile#2046]
### Description of problem
If I scan a QR code and then back out of it or go home, then go scan a **different** QR code, it brings up the proof for the originally scanned one.

### Expected behavior
When I scan the QR code it brings up the proof requested for the QR I scanned, not one sitting in the app already

### Steps to reproduce
Attached video. Described as followed:

- 3 windows
  1. Browser with wallet showcase at CSB demo proof step
  2. Screen share of Android
  3. The a2a **DEV** environment using VCAuth-N

- 00:00 to 00:19. Show what the a2a dev proof **should** look like. See LSBC-demo-issuer-DEV. Go back to starting state
- 00:20 to 00:38. Scan the showcase CSB demo proof. See Law Society of BC (DEMO) cred.
- 00:38. Go back to home, don't do anything with the proof
- 00:45 to 01:00. Scan the a2a dev proof. Now see that it brings up the wrong proof. It's asking for the showcase one that I went back from.
- After that share the proof and it actions the showcase page.
Happens on my iPad as well in the same way but I can't record that.

### Screenshots and/or log output

https://github.com/bcgov/bc-wallet-mobile/assets/17445138/bfe49b6f-165b-49b7-8d20-04f6170e0c2d

### Environment
- [x] Occurs on Android
- [x] Occurs on iOS

Build #: 
**Android:** 1774
**iPad**: 1760

Android Device Model: Pixel 8

iOS Device Model: iPad Air

### Workaround
Cancel the proof after noticing it's the wrong one and scan again

### Severity
- [ ] High
- [ ] Medium
- [ ] Low

**Comment by cvarjao** [comment]
I am wondering if maybe we are not quite using [Navigation.navigate()](https://reactnavigation.org/docs/7.x/navigation-object#navigate) and navigation stacks consistently, and we end up with "ghost"/previous stacks.

**Comment by loneil** [comment]
As discussed with Clecio/Jason this seems to be related to connectionless Proof Requests only(?)

Example body of what scanning a VCAuth-N QR code links to

```
{
    "@id": "d32af120-a7b9-4824-979f-da0bdc198897",
    "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/present-proof/1.0/request-presentation",
    "request_presentations~attach": [
        {
            "@id": "libindy-request-presentation-0",
            "mime-type": "application/json",
            "data": {
                "base64": "eyJub25jZSI6ICI0OTE3OTYzNjg3ODE0MTg1NzM2NjIwMzkiLCAibmFtZSI6ICJwcm9vZl9yZXF1ZXN0ZWQiLCAidmVyc2lvbiI6ICIwLjAuMSIsICJyZXF1ZXN0ZWRfYXR0cmlidXRlcyI6IHsicmVxX2F0dHJfMCI6IHsibmFtZXMiOiBbIlBQSUQiLCAiR2l2ZW4gTmFtZSIsICJTdXJuYW1lIiwgIk1lbWJlciBTdGF0dXMiLCAiTWVtYmVyIFN0YXR1cyBDb2RlIl0sICJyZXN0cmljdGlvbnMiOiBbeyJzY2hlbWFfbmFtZSI6ICJNZW1iZXIgQ2VydGlmaWNhdGUiLCAic2NoZW1hX3ZlcnNpb24iOiAiMS4wLjEiLCAiaXNzdWVyX2RpZCI6ICJSem5ZRlBWaEhwWVpnc240SHUzU3RWIn0sIHsic2NoZW1hX25hbWUiOiAiTWVtYmVyIENlcnRpZmljYXRlIiwgInNjaGVtYV92ZXJzaW9uIjogIjEuMC4xIiwgImlzc3Vlcl9kaWQiOiAiRFpCYUhnaEtzVkhjSm9pd2t5a0czciJ9LCB7InNjaGVtYV9uYW1lIjogIk1lbWJlciBDYXJkIiwgInNjaGVtYV92ZXJzaW9uIjogIjEuNS4xIiwgImlzc3Vlcl9kaWQiOiAiVVVIQTNva25wcnZLcnBhN2E2c25jSyJ9LCB7InNjaGVtYV9uYW1lIjogIk1lbWJlciBDYXJkIiwgInNjaGVtYV92ZXJzaW9uIjogIjEuNS4xIiwgImlzc3Vlcl9kaWQiOiAiQXVKcmlnS1FHUkxKYWpLQWViVGdXdSJ9LCB7InNjaGVtYV9uYW1lIjogIk1lbWJlciBDYXJkIiwgInNjaGVtYV92ZXJzaW9uIjogIjEuNS4xIiwgImlzc3Vlcl9kaWQiOiAiNHhFNjhiNlM1VlJGcktNTUcxVTk1TSJ9XSwgIm5vbl9yZXZva2VkIjogeyJmcm9tIjogMTcyMDEyODM0OCwgInRvIjogMTcyMDEyODM0OH19LCAicmVxX2F0dHJfMSI6IHsibmFtZXMiOiBbImZhbWlseV9uYW1lIiwgImdpdmVuX25hbWVzIl0sICJyZXN0cmljdGlvbnMiOiBbeyJzY2hlbWFfbmFtZSI6ICJ1bnZlcmlmaWVkX3BlcnNvbiIsICJzY2hlbWFfdmVyc2lvbiI6ICIwLjEuMCIsICJpc3N1ZXJfZGlkIjogIlVpNkhBMzZGdk44M2NFdG1ZWUh4cm4ifSwgeyJzY2hlbWFfbmFtZSI6ICJ1bnZlcmlmaWVkX3BlcnNvbiIsICJzY2hlbWFfdmVyc2lvbiI6ICIwLjQuMCIsICJpc3N1ZXJfZGlkIjogIk5Dd0d3RHJ6YlpFcWVzWVF1bW16V1cifSwgeyJzY2hlbWFfbmFtZSI6ICJQZXJzb24iLCAic2NoZW1hX3ZlcnNpb24iOiAiMS4wIiwgImlzc3Vlcl9kaWQiOiAiWHBnZVFhOTNlWnZHU1pCWmVmM1BIbiJ9LCB7InNjaGVtYV9uYW1lIjogIlBlcnNvbiIsICJzY2hlbWFfdmVyc2lvbiI6ICIxLjAiLCAiaXNzdWVyX2RpZCI6ICI3eGpmYXdjbnlUVWNkdVdWeXNMd3c1In0sIHsic2NoZW1hX25hbWUiOiAiUGVyc29uIiwgInNjaGVtYV92ZXJzaW9uIjogIjEuMCIsICJpc3N1ZXJfZGlkIjogIlJHaldiVzFleWNQN0ZyTWY0UUp2WDgifV0sICJub25fcmV2b2tlZCI6IHsiZnJvbSI6IDE3MjAxMjgzNDgsICJ0byI6IDE3MjAxMjgzNDh9fX0sICJyZXF1ZXN0ZWRfcHJlZGljYXRlcyI6IHt9fQ=="
            }
        }
    ],
    "comment": null,
    "~service": {
        "recipientKeys": [
            "6bQ4cCjJkaeWgazXauu8fbFEP5jaZE4ncv4ETu7537Ha"
        ],
        "routingKeys": null,
        "serviceEndpoint": "https://vc-authn-oidc-agent-dev.apps.silver.devops.gov.bc.ca"
    }
}
```

**Comment by jleach** [comment]
@loneil I tracked the root cause down. Maybe its been this way for a while, maybe it changed in Credo 0.5.0, but when we get a connectionless proof we get the `threadId` from the invitation and then use this to match the corresponding proof in Credo. The issue is that the `threadId` in the proof does not match the `threadId` in the invitation. What does match is the `parentThreadId` so I believe this is what we need to use to find the correct / matching proof.

**Comment by knguyenBC** [comment]
I tested and see that this issue persists. Based on the comments, was there or will there be anything done to correct this bug?

**Comment by bryce-mcmath** [comment]
Appears to be fixed in the latest 1.0.18, I think we may have just moved it to QA a bit early.

**Comment by loneil** [comment]
Looks to be working for me. I can't reproduce the situation any more. Using the new 1792 1.0.18 on Android at least.
